src/tests/ftest/control/dmg_telemetry_io_basic.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from avocado.core.exceptions import TestFail
+
+from ior_test_base import IorTestBase
+from telemetry_test_base import TestWithTelemetry
+
+
+class TestWithTelemetryIOBasic(IorTestBase,TestWithTelemetry):
+    # pylint: disable=too-many-ancestors
+    """Test telemetry engine io basic metrics.
+
+    :avocado: recursive
+    """
+
+    def verify_io_test_metrics(self, io_test_metrics, metrics_data, threshold):
+        """ Verify telemetry io metrics from metrics_data.
+
+        Args:
+            io_test_metrics (list): list of telemetry io metrics.
+            metrics_data (dict): a dictionary of host keys linked to a
+                                 list of io metric names.
+            threshold (int): test io metrics threshold.
+
+        """
+        for host in self.hostlist_servers:
+            self.log.info("==Host: %s", host)
+            status = 0
+            for name in io_test_metrics:
+                self.log.info("  --telemetry metric: %s", name)
+                for key in sorted(metrics_data):
+                    m_data = metrics_data[key]
+                    self.log.info(
+                        "  %12s: %s",
+                        "Initial    " if key == 0 else "Test Loop {}".format(
+                            key), metrics_data[key][host][name])
+
+                    #Detail for each test io metrics threshold to be updated
+                    if m_data[host][name] < threshold:
+                        self.log.info(
+                            "#Host %s, test io metrics %s, value %s "
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions
(pylint-too-many-format-args) Too many arguments for format string